src/utils/AddressUtils.ts

```diff
@@ -270,17 +228,34 @@ export class Address {
 // Subclass of address which strictly deals with 20-byte addresses. These addresses are guaranteed to be valid EVM addresses, so `toAddress` will always succeed.
 export class EvmAddress extends Address {
   // On construction, validate that the address can indeed be coerced into an EVM address. Throw immediately if it cannot.
-  constructor(rawAddress: Uint8Array) {
+  protected constructor(rawAddress: Uint8Array) {
     super(rawAddress);
     const hexString = utils.hexlify(rawAddress);
-    if (!this.isValidEvmAddress()) {
+    if (!isValidEvmAddress(hexString)) {
       throw new Error(`${hexString} is not a valid EVM address`);
     }
   }
 
-  // Override `toAddress` to return the 20-byte representation address.
-  override toAddress(): string {
-    return this.toEvmAddress();
+  // Formats the address as a valid EVM address. This function should never throw as constructor-check will have ensured we have a valid Evm address
+  formatAsChecksummedEvmAddress(): string {
```

Comment:
idk if it's necessary to have this as a separate function - for me it's OK to have it under the "to native address" function.